src/V3DynamicScheduler.cpp

```diff
@@ -0,0 +1,1059 @@
+    }
+    virtual void visit(AstVar* nodep) override {
+        if (!m_classp) return;
+        if (nodep->dtypep()->basicp() && nodep->dtypep()->basicp()->isEventValue()) {
+            m_resetStmtsp = AstNode::addNext(
+                m_resetStmtsp,
+                new AstCStmt{nodep->fileline(), "vlSymsp->__Vm_eventDispatcher.cancel("
+                                                    + nodep->nameProtect() + ");\n"});
+        }
+    }
+    virtual void visit(AstCFunc* nodep) override {
+        if (nodep->isDestructor())
+            m_destructor = nodep;
+        else if (nodep->isConstructor())
+            m_constructor = nodep;
+    }
+
+    //--------------------
+    virtual void visit(AstNode* nodep) override { iterateChildren(nodep); }
+
+public:
+    // CONSTRUCTORS
+    explicit DynamicSchedulerClassEventVisitor(AstNetlist* nodep) { iterate(nodep); }
+    virtual ~DynamicSchedulerClassEventVisitor() override {}
+};
+
+//######################################################################
+// DynamicScheduler class functions
+
+void V3DynamicScheduler::transform(AstNetlist* nodep) {
+    UINFO(2, __FUNCTION__ << ": " << endl);
+    UINFO(2, "  Transform Intra Assign Delays...\n");
+    { DynamicSchedulerIntraAssignDelayVisitor{nodep}; }
+    V3Global::dumpCheckGlobalTree("dsch_intra", 0, v3Global.opt.dumpTreeLevel(__FILE__) >= 6);
+    UINFO(2, "  Mark Dynamic...\n");
+    DynamicSchedulerMarkDynamicVisitor visitor(nodep);  // Keep it around to keep user data
+    V3Global::dumpCheckGlobalTree("dsch_mark_dyn", 0, v3Global.opt.dumpTreeLevel(__FILE__) >= 6);
+    UINFO(2, "  Add AstResumeTriggered...\n");
+    auto fl = nodep->fileline();
+    auto* activep = new AstActive{fl, "resumeTriggered",
+                                  new AstSenTree{fl, new AstSenItem{fl, AstSenItem::Combo()}}};
+    activep->sensesStorep(activep->sensesp());
+    activep->addStmtsp(new AstResumeTriggered{
+        fl, visitor.getDlyEvent() ? new AstVarRef{fl, visitor.getDlyEvent(), VAccess::WRITE}
+                                  : nullptr});
+    nodep->topScopep()->scopep()->addActivep(activep);
+    UINFO(2, "  Move Forked Processes to New Functions...\n");
+    { DynamicSchedulerForkVisitor{nodep}; }
+    V3Global::dumpCheckGlobalTree("dsch_forks", 0, v3Global.opt.dumpTreeLevel(__FILE__) >= 6);
+    UINFO(2, "  Add Edge Events...\n");
+    { DynamicSchedulerCreateEdgeEventsVisitor{nodep}; }
+    V3Global::dumpCheckGlobalTree("dsch_events", 0, v3Global.opt.dumpTreeLevel(__FILE__) >= 6);
+    UINFO(2, "  Add Edge Event Triggers...\n");
+    { DynamicSchedulerContinuousAssignVisitor{nodep}; }
+    { DynamicSchedulerAddTriggersVisitor{nodep}; }
+    V3Global::dumpCheckGlobalTree("dsch", 0, v3Global.opt.dumpTreeLevel(__FILE__) >= 3);
+}
+
+void V3DynamicScheduler::classCleanup(AstNetlist* nodep) {
+    { DynamicSchedulerClassEventVisitor{nodep}; }
+    V3Global::dumpCheckGlobalTree("dsch_classes", 0, v3Global.opt.dumpTreeLevel(__FILE__) >= 6);
+}
```

Comment:
If there's no code shared with other visitors (I think not) please make a separate V3DynamicClass.cpp.